iceoryx_utils/source/posix_wrapper/timer.cpp

```diff
@@ -254,21 +306,26 @@ cxx::expected<TimerError> Timer::OsTimer::start(const RunMode runMode, const Cat
         interval.it_interval.tv_nsec = 0;
     }
 
+    auto& handle = OsTimer::s_callbackHandlePool[m_callbackHandleIndex];
+    // setting m_isTimerActive after timer_settime could lead to false negatives in a check which decides whether
+    // the callback should be executed,
+    // setting it beforehand leads to false positives which can be dealt with and avoid this problem
+    auto wasActive = handle.m_isTimerActive.exchange(true, std::memory_order_relaxed);
+    handle.m_catchUpPolicy = catchUpPolicy;
+
     // Set the timer
     auto result = cxx::makeSmartC(
         timer_settime, cxx::ReturnMode::PRE_DEFINED_ERROR_CODE, {-1}, {}, m_timerId, 0, &interval, nullptr);
 
     if (result.hasErrors())
     {
+        // undo optimistically setting m_isTimerActive before
+        // not entirely safe against concurrent starts, we cannot detect these in this way
+        // needs extensive refactoring to achieve this (e.g. start and stop with mutex protection?)
+        handle.m_isTimerActive.exchange(wasActive, std::memory_order_relaxed);
         return createErrorFromErrno(result.getErrNum());
     }
 
-    auto& handle = OsTimer::s_callbackHandlePool[m_callbackHandleIndex];
-
-    handle.m_catchUpPolicy = catchUpPolicy;
```

Comment:
this has either to be set here, maybe made atomic, or removed from the function signature